allennlp/data/token_indexers/pretrained_transformer_pretokenized_indexer.py

```diff
@@ -0,0 +1,95 @@
+from typing import Dict, List, Tuple
+import logging
+
+from overrides import overrides
+import torch
+
+from allennlp.common.util import pad_sequence_to_length
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data.tokenizers.token import Token
+from allennlp.data.token_indexers import PretrainedTransformerIndexer, TokenIndexer
+from allennlp.data.token_indexers.token_indexer import IndexedTokenList
+from allennlp.data.tokenizers import PretrainedTransformerTokenizer
+
+logger = logging.getLogger(__name__)
+
+
+@TokenIndexer.register("pretrained_transformer_pretokenized")
+class PretrainedTransformerPretokenizedIndexer(PretrainedTransformerIndexer):
+    """
+    Use this indexer when input comes from pre-tokenized text and therefore
+    `PretrainedTransformerTokenizer` was not used in the dataset loader. It tokenizes each token
+    into wordpieces independently and concatenate them back together. Use it along with
+    `PretrainedTransformerPretokenizedEmbedder`.
+    """
+
+    def __init__(
+        self, model_name: str, namespace: str = "tags", token_min_padding_length: int = 0
+    ) -> None:
+        super().__init__(model_name, namespace, token_min_padding_length)
+        # add_special_tokens=False sicne we don't want wordpieces to be surrounded by special tokens
+        self._allennlp_tokenizer = PretrainedTransformerTokenizer(
+            model_name, add_special_tokens=False
+        )
+
+    @overrides
+    def tokens_to_indices(self, tokens: List[Token], vocabulary: Vocabulary) -> IndexedTokenList:
+        orig_token_mask = [1] * len(tokens)
+        tokens, offsets = self._intra_word_tokenize(tokens)
+
+        # {"token_ids": ..., "mask": ...}
+        output = super().tokens_to_indices(tokens, vocabulary)
+
+        # self._intra_word_tokenize() does not insert special tokens, so we need to do it here
+        output["token_ids"] = self._tokenizer.build_inputs_with_special_tokens(output["token_ids"])
+        output["mask"] = orig_token_mask
+        output["offsets"] = [
+            (start + self._num_added_start_tokens, end + self._num_added_start_tokens)
+            for start, end in offsets
+        ]
+        output["wordpiece_mask"] = [1] * len(output["token_ids"])
+        return output
+
+    @overrides
+    def get_empty_token_list(self) -> IndexedTokenList:
+        output = super().get_empty_token_list()
+        output["offsets"] = []
+        output["wordpiece_mask"] = []
+        return output
+
+    @overrides
+    def as_padded_tensor_dict(
+        self, tokens: IndexedTokenList, padding_lengths: Dict[str, int]
+    ) -> Dict[str, torch.Tensor]:
+        offsets_tokens = tokens.pop("offsets")
+        offsets_padding_lengths = padding_lengths.pop("offsets")
+
+        tensor_dict = super().as_padded_tensor_dict(tokens, padding_lengths)
```

Comment:
What kind of tests are you proposing that I should add here?

Reply:
I hadn't looked at the tests when I wrote this, I was just thinking out loud.  The one end-to-end test that you have is maybe sufficient, though it could use some more detail.  I'll comment there.